plaso/parsers/onedrive.py

```diff
@@ -0,0 +1,441 @@
+    """Parses the ObfuscatedStringMap.txt file.
+
+    Args:
+      file_entry (dfvfs.FileEntry): the OneDrive Log file entry.
+      parser_mediator  (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+
+    Returns:
+      dict[str, str]: a mapping of an obfuscated string value to its true string
+          value.  None if the ObfuscatedSTringMap.txt file could not be found.
+    """
+    obfuscated_string_map_entry = self._OpenFileInSameDirectory(
+        file_entry, self._OBFUSCATED_STRING_MAP_FILE, parser_mediator)
+
+    if not obfuscated_string_map_entry:
+      return None
+
+    obfuscated_string_map_file = obfuscated_string_map_entry.GetFileObject()
+
+    obfuscated_string_map = {}
+    with text_file.TextFile(obfuscated_string_map_file,
+        encoding='utf-16-le') as obfuscated_text_file:
+      for line in obfuscated_text_file.readlines():
+        cleaned_line = line.rstrip().split('\t')
+        if len(cleaned_line) != 2:
+          continue
+        if cleaned_line[0] in obfuscated_string_map:
+          continue
+        obfuscated_string_map[cleaned_line[0]] = cleaned_line[1]
+    return obfuscated_string_map
+
+  def _ParseGeneralKeyStore(self, file_entry, parser_mediator):
+    """Parses the general.keystore file.
+
+    Args:
+      file_entry (dfvfs.FileEntry): the OneDrive Log file entry.
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+
+    Returns:
+      bytes: the key, or None if the general.keystore file could not be found or
+          parsed.
+    """
+    general_keystore_entry = self._OpenFileInSameDirectory(
+        file_entry, self._GENERAL_KEYSTORE_FILE, parser_mediator)
+
+    if not general_keystore_entry:
+      return None
+
+    general_keystore_file_object = general_keystore_entry.GetFileObject()
+    general_keystore_data = general_keystore_file_object.read()
+    json_data = json.loads(general_keystore_data)
+
+    if not json_data:
+      parser_mediator.ProduceExtractionWarning(
+          'Unable to parse OneDrive keystore file {0:s}'.format(
+              general_keystore_entry.name))
+      return None
+
+    key = json_data[0].get('Key', None)
+    decoded_key = base64.b64decode(key)
+
+    if len(decoded_key) != 32:
+      return None
+
+    return decoded_key
+
+  def _DeobfuscateStrings(self, extracted_strings, obfuscated_string_map):
+    """Deobfuscate strings extracted from OneDrive logs.
+
+    Args:
+      extracted_strings (list[str]): the list of strings extracted from OneDrive
+          log files.
+      obfuscated_string_map (dict[str, str]): the parsed obfuscated string map.
+
+    Returns:
+      list[str]: the list of deobfuscated strings.
+    """
+    if not obfuscated_string_map:
+      return extracted_strings
+
+    delimiters = r'(' + r'|'.join(self._DELIMITERS) + r')'
```

Comment:
Why the sudden style change? Why not:
`delimiters = '({0:s})'.format('|'.join(self._DELIMITERS))`

Reply:
Thanks for the commit/update.